tests/MongoDB.Driver.Core.Tests/Core/Operations/RetryableWriteOperationExecutorTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers;
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Connections;
+using MongoDB.Driver.Core.Operations;
+using MongoDB.Driver.Core.Servers;
+using Moq;
+using Xunit;
+
+namespace MongoDB.Driver.Core.Tests.Core.Operations
+{
+    public class RetryableWriteOperationExecutorTests
+    {
+        [Theory]
+        [InlineData(false, false, false, false)]
+        [InlineData(false, false, true, false)]
+        [InlineData(false, true, false, false)]
+        [InlineData(false, true, true, false)]
+        [InlineData(true, false, false, false)]
+        [InlineData(true, false, true, false)]
+        [InlineData(true, true, false, true)]
+        [InlineData(true, true, true, false)]
+        public void DoesContextAllowRetries_should_return_expected_result(
+            bool retryRequested,
+            bool areRetryableWritesSupported,
+            bool isInTransaction,
+            bool expectedResult)
+        {
+            var context = CreateContext(retryRequested, areRetryableWritesSupported, isInTransaction);
+
+            var result = RetryableWriteOperationExecutorReflector.DoesContextAllowRetries(context);
+
+            object should = result.Should().Be(expectedResult);
+        }
+
+        [Theory]
+        [InlineData(false, false, true)]
```

Comment:
Nit: I don't think this case is needed, since isAcknowledged is ignored when withWriteConcern is false.